RT 2.1 and RT 2.2 base_adjacencies_test.go , lsp_updates_test.go

cprabha opened the pull request:
Hi,

Below are the changes done for RT 2.1 and RT 2.2 scripts. 

session.go:

1. Added code to cleanup protocol config before running test to remove any stale isis state and counters. 

RT 2.1 base_adjacencies_test.go:

1.Area address telemetry contains both DUT and ATE Area addresses. 
  Test expects DUT Area Address and ATE Area address to be in specific order. 
  Added modification to verify the list of AreaAddress in unordered way.

2. Deadline interval was 5 seconds , it is changed to 1 minute . 
   Since multiple RPC calls are being placed in quick succession, increasing the grpc
   response time to accomodate the response time for all the RPC's 

RT 2.2 lsp_updates_test.go:

1. In telemetry verification, extra timeout is needed since multiple RPCs are executed in quick succession

Note: Same changes are applied to OTG scripts.

Thanks,
Prabha

sachendras commented on feature/experimental/isis/ate_tests/base_adjacencies_test/base_adjacencies_test.go:
```diff
@@ -60,12 +69,12 @@ func TestBasic(t *testing.T) {
 		t.Fatalf("IS-IS failed to configure: %v", err)
 	}
 	// There might be lag between when the instance name is set and when the
 	// other parameters are set; we expect the total lag to be under 5s
```
Comment doesnt seem to match with what you are proposing here. Can you please update?

cprabha replied:
updated , Thank you.

sachendras commented on feature/experimental/isis/ate_tests/base_adjacencies_test/base_adjacencies_test.go:
```diff
@@ -192,12 +201,11 @@ func TestBasic(t *testing.T) {
 	// Allow 1s of lag between adjacency appearing and all data being populated
```
Again, comment dont seem to match what you are proposing here. Can you please update?

cprabha replied:
updated , Thank you.

cprabha commented:
Hi,

I have updated OTG files with similar changes. 

Thanks,
Prabha

sachendras commented:
Hello cprabha

IIUC, you are saying that the changes in this PR should now pass provided the IXIA has a patch to fix an existing bug as per 
https://github.com/openconfig/featureprofiles/issues/808. Can you please provide more info on what was the patch version for IXIA? We require that to ensure our Testbeds are also updated accordingly.

Thanks

cprabha commented:
Hi Sachendra,

Yes, issue is with ixia sending CSN/LSP packets before adjacency formation.  DUT will discard these packets as adjacency is not up and script is expecting for 0 discarded packets and hence fails.

Subhajit Pal ***@***.******@***.***>; Anush Varada ***@***.******@***.***> provided patch which prevent sending CSN/LSP packets before adjacency formation. Script works fine with this patch.

Below are checks where it fails with ixia appserver when patch not applied .


check.Equal(pCounts.Csnp().Dropped().State(), uint32(0)),

check.Equal(pCounts.Lsp().Dropped().State(), uint32(0)),
check.NotEqual(pCounts.Csnp().Processed().State(), uint32(0)),


Thanks,
Prabha



Juniper Business Use Only
From: sachendras ***@***.***>
Sent: Monday, January 23, 2023 5:45 PM
To: openconfig/featureprofiles ***@***.***>
Cc: Prabha Chethan ***@***.***>; Author ***@***.***>
Subject: Re: [openconfig/featureprofiles] RT 2.1 and RT 2.2 base_adjacencies_test.go , lsp_updates_test.go (PR #792)

[External Email. Be cautious of content]


Hello cprabha

IIUC, you are saying that the changes in this PR should now pass provided the IXIA has a patch to fix an existing bug as per
#808<https://urldefense.com/v3/__https:/github.com/openconfig/featureprofiles/issues/808__;!!NEt6yMaO-gk!HvRKULjq0unIzI12G2DAtyKlEpI6SwzjT0bwcb3WFwURhr3vvQhEZ77twDXswNWcO0fRXjp5-nwJF1l8jX0zMLeu$>. Can you please provide more info on what was the patch version for IXIA? We require that to ensure our Testbeds are also updated accordingly.

Thanks

-
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https:/github.com/openconfig/featureprofiles/pull/792*issuecomment-1401294872__;Iw!!NEt6yMaO-gk!HvRKULjq0unIzI12G2DAtyKlEpI6SwzjT0bwcb3WFwURhr3vvQhEZ77twDXswNWcO0fRXjp5-nwJF1l8jcK7W3Hk$>, or unsubscribe<https://urldefense.com/v3/__https:/github.com/notifications/unsubscribe-auth/AHCEB5SNZRQ6LQDRLUYFWCDWT4X2HANCNFSM6AAAAAASC3DVOQ__;!!NEt6yMaO-gk!HvRKULjq0unIzI12G2DAtyKlEpI6SwzjT0bwcb3WFwURhr3vvQhEZ77twDXswNWcO0fRXjp5-nwJF1l8jS0FAglh$>.
You are receiving this because you authored the thread.Message ID: ***@***.******@***.***>>

sachendras commented:
Can you please help us understand what version of IXIA has this fix? Version 5.30 as mentioned in the offline spreadsheet seems not valid as it is obsolete.

joannc2022 commented:
The Ixia fix is present in recent release 9.30 OS @ https://support.ixiacom.com/version/ixnetwork-930

xw-g commented on feature/experimental/isis/ate_tests/base_adjacencies_test/base_adjacencies_test.go:
```diff
@@ -61,8 +70,8 @@ func TestBasic(t *testing.T) {
-	// other parameters are set; we expect the total lag to be under 5s
-	deadline := time.Now().Add(time.Second * 5)
+	// other parameters are set; we expect the total lag to be under one minute
+	deadline := time.Now().Add(time.Minute)
```
why one min?

cprabha replied:
each rpc call needs 3 seconds duration and hence it is calculated as minute based on number of rpcs . At line number 133 to 147, we have 1 rpc calls in single for loop.  Hence came up with 1 minute duration for deadline variable.

xw-g commented on feature/experimental/isis/ate_tests/base_adjacencies_test/base_adjacencies_test.go:
```diff
@@ -204,15 +213,14 @@ func TestBasic(t *testing.T) {
 		t.Fatalf("No IS-IS adjacency formed: %v", err)
 	}
-	// Allow 1s of lag between adjacency appearing and all data being populated
+	// Allow 1 Minute of lag between adjacency appearing and all data being populated
```
same thing.

cprabha replied:
It is calculated based on number of rpcs in this block of code

xw-g commented on feature/experimental/isis/ate_tests/base_adjacencies_test/base_adjacencies_test.go:
```diff
@@ -239,6 +247,14 @@ func TestBasic(t *testing.T) {
 				}
 			})
 		}
+		status := gnmi.Get(t, ts.DUT, adj.AreaAddress().State())
```
comments.

cprabha replied:
I will add comments and commit the code

xw-g commented on feature/experimental/isis/ate_tests/base_adjacencies_test/base_adjacencies_test.go:
```diff
@@ -250,7 +266,7 @@ func TestBasic(t *testing.T) {
 		//   rest of the counters is pointless - none of them will change if we
 		//   haven't been exchanging IS-IS messages.
-		deadline = time.Now().Add(time.Second * 5)
+		deadline = time.Now().Add(time.Second * 15)
```
same thing, why 15 sec.

cprabha replied:
it is calculated based on number of rpc calls, each rpc call needs 3 secs.

xw-g commented on feature/experimental/isis/ate_tests/base_adjacencies_test/base_adjacencies_test.go:
```diff
@@ -262,7 +278,7 @@ func TestBasic(t *testing.T) {
 			})
 		}
-		deadline = time.Now().Add(time.Second)
+		deadline = time.Now().Add(time.Minute)
```
same, why 1 min.

cprabha replied:
each rpc needs 3 secs, if multiple rpc calls are in this block of code then we need to increase that duration

xw-g commented on feature/experimental/isis/ate_tests/internal/session/session.go:
```diff
@@ -256,7 +256,15 @@ func (s *TestSession) PushDUT(ctx context.Context) error {
 	dutConf := s.DUTConf.GetOrCreateNetworkInstance(*deviations.DefaultNetworkInstance).GetOrCreateProtocol(PTISIS, ISISName)
-	_, err := ygnmi.Replace(ctx, s.DUTClient, ProtocolPath().Config(), dutConf)
+
+	// Clear ISIS Protocol
```
the purpose of clearing ISIS protocol.

cprabha replied:
Clearing ISIS protocol makes all counters zero, test checks for these counters in beginning of the test.

xw-g commented on feature/experimental/isis/ate_tests/lsp_updates_test/lsp_updates_test.go:
```diff
@@ -50,7 +50,7 @@ func TestOverloadBit(t *testing.T) {
 	ts.MustAdjacency(t)
 	setBit := isisPath.Global().LspBit().OverloadBit().SetBit()
-	deadline := time.Now().Add(time.Second)
+	deadline := time.Now().Add(time.Second * 10)
```
why 10s?

cprabha replied:
1 second is too short, test might fail due to deadline exceeded if timely reply is not received at controller.

xw-g commented on feature/experimental/isis/ate_tests/lsp_updates_test/lsp_updates_test.go:
```diff
@@ -76,7 +76,7 @@ func TestOverloadBit(t *testing.T) {
 		t.Error(err)
 	}
-	if err := check.Equal(setBit.State(), true).AwaitFor(time.Second, ts.DUTClient); err != nil {
+	if err := check.Equal(setBit.State(), true).AwaitFor(time.Second*15, ts.DUTClient); err != nil {
```
same.

cprabha replied:
one second is not sufficient, test will fail with deadline exceeded error

xw-g commented on feature/experimental/isis/ate_tests/lsp_updates_test/lsp_updates_test.go:
```diff
@@ -99,7 +99,7 @@ func TestMetric(t *testing.T) {
 	metric := session.ISISPath().Interface(ts.DUTPort1.Name()).Level(2).
 		Af(oc.IsisTypes_AFI_TYPE_IPV4, oc.IsisTypes_SAFI_TYPE_UNICAST).Metric()
-	if err := check.Equal(metric.State(), uint32(100)).AwaitFor(time.Second, ts.DUTClient); err != nil {
+	if err := check.Equal(metric.State(), uint32(100)).AwaitFor(time.Second*15, ts.DUTClient); err != nil {
```
same.

cprabha replied:
1 second is too short.

xw-g commented on feature/experimental/isis/otg_tests/internal/session/session.go:
```diff
@@ -51,7 +51,7 @@ const (
 	DUTSysID       = "1920.0000.2001"
 	ATESysID       = "640000000001"
 	// TODO: Change the name to DEFAULT
```
remove the comment.

cprabha replied:
I will remove and commit code

xw-g commented on feature/experimental/isis/otg_tests/internal/session/session.go:
```diff
@@ -255,7 +255,14 @@ func (s *TestSession) PushDUT(ctx context.Context) error {
 	}
 	dutConf := s.DUTConf.GetOrCreateNetworkInstance(*deviations.DefaultNetworkInstance).GetOrCreateProtocol(PTISIS, ISISName)
-	_, err := ygnmi.Replace(ctx, s.DUTClient, ProtocolPath().Config(), dutConf)
+	// Clear ISIS Protocol
```
same thing for OTG.

cprabha replied:
We need to clear existing protocol and counters /stats will be resetted before the test starts.

xw-g commented on feature/experimental/isis/ate_tests/base_adjacencies_test/base_adjacencies_test.go:
```diff
@@ -49,6 +49,16 @@ func EqualToDefault[T any](query ygnmi.SingletonQuery[T], val T) check.Validator
 }
 
+// checkSliceContains checks if given string s1 is found in string array s2.
+func checkSliceContains(s1 string, s2 []string) bool {
```
Let's try to enhance the check package, instead of adding one time helpers here.

cprabha replied:
removed this code and changes are committed.

xw-g commented:
LGTM. once internal presubmit test passes. 

@sachendras I'll let you drive the merge based on internal presubmit test.

sachendras commented:
Internal b/262109675 for Google

sachendras commented:
/gcburn

sachendras commented:
/gcburn

sachendras commented:
/gcbrun
